### PR TITLE
added inventory/ folder

### PR DESCRIPTION
**Pull request by pescobar**
having a playbook top folder `inventory/` which also includes `inventory/group_vars/` and `inventory/host_vars/` provides a cleaner directory layout

the provided `ansible.cfg` in the playbook top folder avoid the need of always using `ansible -i`. Still you can use `ansible -i` and it will take precedence over `ansible.cfg`

### CODE REVIEW

**Comment by bedroge** [comment]
Could you pull in the latest changes from the master branch (i.e. from your other PR)? Then I can give this a try.

**Comment by pescobar** [comment]
@bedroge I have merged master branch from upstream repo

File: inventory/group_vars/cvmfsstratum1servers [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+
+#cvmfs_srv_device: /dev/sdb
```
Comment (bedroge) [inline inventory/group_vars/cvmfsstratum1servers:3]: This should go the host_vars, but we also can do that later on / in a separate PR.

**Comment by bedroge** [comment]
Made some updates to your PR: updated the README, and added a `host_vars` directory with some examples in it. 

There's still some configuration in `hosts.examples` regarding IP addresses of proxies/clients. I don't think they should go to host_vars, since this applies to multiple machines (clients/proxies). They could go in a group_vars, but then I don't really like that everyone has to change files that are tracked by git. So I left them in the hosts file for now; where do you think they should go?

**Comment by pescobar** [comment]
if we follow the same approach it should go in `group_vars/cvmfslocalproxies` but I see your point.

First idea that comes to my mind is to define a new config file named something like `local_config_options.yml` which is not tracked by git. Inside this config file users could define all the site specific settings e.g. this var (same name but starting with `local_`):

```
local_cvmfs_localproxy_allowed_clients = ["192.168.0.0/12", "10.0.0.15"]
```

and then in file `group_vars/cvmfslocalproxies` which is provided by the git repo we could have:

```
cvmfs_localproxy_allowed_clients = "{{ local_cvmfs_localproxy_allowed_clients }}"
```

Then in the README we can explain that all the site specific settings should be defined in `local_config_options.yml` and for those who need to use custom settings they should execute the playbooks like `ansible-playbook -e @local_config_options.yml xxxxx.yml`

I am not sure if I am overengineering though. What's your opinion?

Another option is that inside `inventory` folder we only provide example files and users are instructed to create their own files adapted to their specific settings. But maybe this is too much work for users and more error prone?

**Comment by bedroge** [comment]
I like the first option. I think most of the stuff in group_vars is pretty much the same for most users running these playbooks, so having one file (or actually two, including the hosts file) with site specific stuff sounds like a good idea. I'll try to change that. Thanks!

**Comment by bedroge** [comment]
What would be the best place to store this `local_config_options.yml`? Also in `group_vars`? And what do we do with the things that I now put in the `host_vars/*.example` files, e.g. the GEO API key? Does it make sense to keep it there, or should we move that to `local_config_options.yml as well`?

**Comment by pescobar** [comment]
what do you think about placing it inside the `inventory` folder? We could use `inventory/local_site_specific_vars.yml` . Personally I like this option. Another alternative is to place the file in the repo top folder. In any case the important bit is to document it properly in the README so users know that this is the right file to define their site specific variables and also explain how to execute the playbooks doing `ansible-playbook -e @path/to/local_site_specific_vars.yml xxxxx.yml`

The GEO API key is site specific? If that the case IMO it should go in this file.

**Comment by bedroge** [comment]
Okay, I changed it a bit more and updated the README. Please have a look :-)
Note that I haven't tested it yet, so there could still be bugs in this new setup.

File: inventory/group_vars/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+local_config_options.yml
```
Comment (pescobar) [inline inventory/group_vars/.gitignore:1]: I think this is not needed?

File: inventory/host_vars/stratum0host.example [hunk 3]
```diff
@@ -0,0 +1 @@
+cvmfs_srv_device: /dev/sdb
```
Comment (pescobar) [inline inventory/host_vars/stratum0host.example:1]: what about adding this to `local_site_specific_vars.yml` ? and then use here something like:
```
cvmfs_srv_device: "{{ local_cvmfs_srv_device | default('/dev/sdb') }}"
```

File: inventory/group_vars/.gitignore (hunk 2 above)
Comment (bedroge) [reply inventory/group_vars/.gitignore]: Ah, yes, I initially put the file in this dir. Just removed the .gitignore.

File: inventory/local_site_specific_vars.yml.example [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# The license key for the Geo API:
+# https://cvmfs.readthedocs.io/en/stable/cpt-replica.html#geo-api-setup
+# For some unclear reason, the Stratum 1 installation fails when this is not set:
+# https://github.com/EESSI/filesystem-layer/issues/2
+cvmfs_geo_license_key: INSERT_YOUR_KEY
```
Comment (pescobar) [inline inventory/local_site_specific_vars.yml.example:5]: nitpicking here but to follow the same convention every site specific variable starts by `local_`

File: inventory/host_vars/stratum0host.example (hunk 3 above)
Comment (bedroge) [reply inventory/host_vars/stratum0host.example:1]: But I think this is machine specific? Maybe a site has a stratum 0 and stratum 1, and for the first they want to use /dev/sdb and for the latter /dev/sdc?

Comment (pescobar) [reply inventory/host_vars/stratum0host.example:1]: ups, I see this is in an example file and will always be a site specific settings so I think it's ok like it is. Please ignore my comment

File: inventory/local_site_specific_vars.yml.example (hunk 4 above)
Comment (pescobar) [reply inventory/local_site_specific_vars.yml.example:5]: same here. it's a example file so it's always site-specific. please ignore the comment. looks perfect

File: inventory/host_vars/stratum0host.example (hunk 3 above)
Comment (pescobar) [reply inventory/host_vars/stratum0host.example:1]: you are totally right. I missed it's an example. IMO it's good to go

File: inventory/local_site_specific_vars.yml.example (hunk 4 above)
Comment (bedroge) [reply inventory/local_site_specific_vars.yml.example:5]: Okay! I'm not completely sure about the local_ anyway: if we want to support that for all variables, we would need extra code in the group_vars/all.yml to pick up these variables.

Comment (pescobar) [reply inventory/local_site_specific_vars.yml.example:5]: I think it makes sense to follow same approach for every var defined in `inventory/local_site_specific_vars.yml` but it's just a small detail. Do it as you prefer

**Comment by bedroge** [comment]
I just tested all playbooks (stratum 0, stratum 1, local proxy, clients) on clean VMs, and it still works fine.

**Comment by pescobar** [comment]
cool :) IMO it's ready to merge and if there is any missing detail it can be polished afterwards in a different PR but with this PR there is a single place in `inventory/` folder to define all the variables in a structured way

**Review by peterstol — APPROVED** [review]
Good idea to define all the variables in a structured way in the inventory folder with an example for the site specific variables.

**Pull request merged**